jobsdb/jobsdb.go

```diff
@@ -873,6 +878,13 @@ func (jd *HandleT) workersAndAuxSetup() {
 // Start starts the jobsdb worker and housekeeping (migration, archive) threads.
 // Start should be called before any other jobsdb methods are called.
 func (jd *HandleT) Start() {
+	jd.lifecycle.mu.Lock()
+	defer jd.lifecycle.mu.Unlock()
+	if jd.lifecycle.started {
```

Comment:
Good addition, it was in my TODO list 😄 

Can we please add a test for this?

Reply:
of course

Reply:
done